src/states/adult/mod.rs

```diff
@@ -547,13 +499,25 @@ impl Base for Adult {
             }
         }
 
+        for msg in mem::replace(&mut self.routing_msg_backlog, Default::default()) {
+            if let Transition::Stay = &transition {
+                match self.handle_filtered_signed_message(msg) {
```

Comment:
calling handle_filtered_signed_message make the node resend the message. At this point in finish_handle_transition we should probably only apply what needs to be done for us.